common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/requests/resetpassword/ResetPasswordChallengeRequest.kt

```diff
@@ -0,0 +1,79 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.providers.nativeauth.requests.resetpassword
+
+import com.google.gson.annotations.SerializedName
+import com.microsoft.identity.common.java.providers.nativeauth.requests.NativeAuthRequest
+import com.microsoft.identity.common.java.util.ArgUtils
+import java.net.URL
+
+/**
+ * Represents a request to the Reset Password /challenge endpoint, and provides a create() function to instantiate the request using the provided parameters.
+ */
+class ResetPasswordChallengeRequest private constructor(
+    override var requestUrl: URL,
+    override var headers: Map<String, String?>,
+    override val parameters: NativeAuthRequestParameters
```

Comment:
Why are these override?

Reply:
`override` keyword is used where the attribute is also present in the base class.